RoslynPluginGenerator/DiagnosticAssemblyScanner.cs

```diff
@@ -31,15 +30,14 @@ public DiagnosticAssemblyScanner(ILogger logger, params string[] additionalSearc
         }
 
         /// <summary>
-        /// Loads all assemblies in the given directory and instantiates Roslyn diagnostic objects - i.e. existing types deriving from
+        /// Loads all of the given assemblies and instantiates Roslyn diagnostic objects - i.e. existing types deriving from
         /// <see cref="DiagnosticAnalyzer"/>
         /// </summary>
         /// <returns>Enumerable with instances of DiagnosticAnalyzer from discovered assemblies</returns>
-        public IEnumerable<DiagnosticAnalyzer> InstantiateDiagnostics(string directoryPath, string language)
+        public IEnumerable<DiagnosticAnalyzer> InstantiateDiagnostics(string language, params string[] files)
         {
             List<DiagnosticAnalyzer> analyzers = new List<DiagnosticAnalyzer>();
-
-            foreach (string assemblyPath in Directory.GetFiles(directoryPath, "*.dll", SearchOption.AllDirectories))
+            foreach (string assemblyPath in files)
```

Comment:
I'm not sure it makes sense to check in both places. It might be better to move the check to the scanner and have it silently ignore non-assembly files.